arches/app/views/api.py

```diff
@@ -2104,3 +2105,307 @@ def get(self, request):
+            )
+        return JSONErrorResponse(_("No json request payload"), status=400)
+
+    @method_decorator(group_required("Application Administrator"))
+    def put(self, request, identifier=None):
+
+        if not identifier:
+            return JSONErrorResponse(
+                _("Spatialview update failed"),
+                _(
+                    "PUT REST request requires a spatialviewid to be provided in the URL"
+                ),
+                status=400,
+            )
+
+        try:
+            json_data = json.loads(request.body.decode("utf-8"))
+        except ValueError:
+            return JSONErrorResponse(
+                _("Invalid JSON data"),
+                _("The Spatialview API was sent invalid JSON"),
+                status=400,
+            )
+
+        if json_data is not None:
+
+            validate_json_data_content_result = self.validate_json_data_content(
+                json_data, identifier
+            )
+            if isinstance(validate_json_data_content_result, JSONErrorResponse):
+                return validate_json_data_content_result
+
+            spatialview = validate_json_data_content_result
+
+            try:
+                spatialview.clean()
+                spatialview.save()
+            except ValidationError as e:
+                return JSONErrorResponse(
+                    _("Validation Error when updating Spatialview"),
+                    e.message,
+                    status=400,
+                )
+
+            return JSONResponse(
+                self.create_json_data_object_from_spatialview(spatialview), status=200
+            )
+        return JSONErrorResponse(
+            _("Spatialview update falied"), _("No json request payload"), status=400
+        )
+
+    @method_decorator(group_required("Application Administrator"))
+    def delete(self, request, identifier=None):
+        if identifier:
+            spatialview = None
+            try:
+                spatialview = models.SpatialView.objects.get(pk=identifier)
+            except ObjectDoesNotExist:
+                return JSONErrorResponse(
+                    _("Spatialview delete failed"),
+                    _("Spatialview does not exist"),
+                    status=404,
+                )
+
+            try:
+                spatialview.delete()
+            except Exception as e:
+                logger.error(e)
+                return JSONErrorResponse(
+                    _("Spatialview delete failed"),
+                    _("An error occurred when trying to delete the spatialview"),
+                    status=500,
+                )
+
+        else:
+            return JSONErrorResponse(
+                _("Spatialview delete falied"),
+                _(
+                    "DELETE REST request requires a spatialviewid to be provided in the URL"
+                ),
+                status=400,
+            )
+        return JSONResponse(status=200)
```

Comment:
Suggest NO_CONTENT instead of OK for a delete

Reply:
done